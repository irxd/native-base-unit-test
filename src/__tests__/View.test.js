import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import {
  NativeBaseProvider, View
} from 'native-base';

describe('<View />', () => {
  it('should render view component properly', () => {
    const { queryByTestId } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <View testID="my-view">Test View</View>
      </NativeBaseProvider>
    )

    expect(queryByTestId('my-view')).not.toBeNull()
  });

  it('should render view props properly', () => {
    const { queryByTestId, toJSON } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <View
          testID="my-view"
          backgroundColor="white"
          borderColor="black"
          marginBottom={4}
        >
          Test View
        </View>
      </NativeBaseProvider>
    )
    
    expect(queryByTestId('my-view').props.style.backgroundColor).toMatch('#FFFFFF');
    expect(queryByTestId('my-view').props.style.borderColor).toMatch('#000000');
    expect(queryByTestId('my-view').props.style.marginBottom).toEqual(16);
    expect(toJSON()).toMatchSnapshot();
  });

  it('view event should working properly', () => {
    const onPressEvent = jest.fn();

    const { queryByTestId, toJSON } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <View
          testID="my-view"
          backgroundColor="white"
          borderColor="black"
          marginBottom={4}
          onPress={onPressEvent}
        >
          Test View
        </View>
      </NativeBaseProvider>
    )
    
    fireEvent.press(queryByTestId('my-view'));
    expect(onPressEvent).toHaveBeenCalled();
    expect(queryByTestId('my-view').props.style.backgroundColor).toMatch('#FFFFFF');
    expect(queryByTestId('my-view').props.style.borderColor).toMatch('#000000');
    expect(queryByTestId('my-view').props.style.marginBottom).toEqual(16);
    expect(toJSON()).toMatchSnapshot();
  });
});
