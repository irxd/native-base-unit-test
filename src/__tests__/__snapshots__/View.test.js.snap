// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View /> should render view props properly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    _text={
      Object {
        "color": "darkText",
      }
    }
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#000000",
        "marginBottom": 16,
      }
    }
    testID="my-view"
  >
    Test View
  </View>
</RNCSafeAreaProvider>
`;

exports[`<View /> view event should working properly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    _text={
      Object {
        "color": "darkText",
      }
    }
    dataSet={Object {}}
    onPress={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#000000",
        "marginBottom": 16,
      }
    }
    testID="my-view"
  >
    Test View
  </View>
</RNCSafeAreaProvider>
`;
